mri/utils/process_SWI

```diff
@@ -0,0 +1,127 @@
+import matplotlib.pyplot as plt
+import numpy as np
+
+#functions
+def virtual_coil_reconstruction(imgs):
+    """
+    Calculate the combination of all the coils thanks to the virtual coil
+    methods as defined:
+    Parameters
+    ----------
+    imgs: np.ndarray
+        The images reconstructed channel by channel [Nch, Nx, Ny, Nz]
+    Returns
+    -------
+    I: np.ndarray
+        The combination of all the channels in a complex valued [Nx, Ny, Nz]
+    """
+    # Compute first the virtual coil
+    nch, nx, ny, nz = imgs.shape
+    weights = np.sum(np.abs(imgs), axis=0)
+    weights[weights==0] = 1e-16
+    phase_reference = np.asarray([np.angle(np.sum(
+        imgs[ch].flatten())) for ch in range(nch)])
+    reference = np.asarray([(imgs[ch] / weights) / np.exp(1j * phase_reference[ch])
+                           for ch in range(32)])
+    virtual_coil = np.sum(reference, axis=0)
+    difference_original_vs_virtual = np.asarray(
+        [np.conjugate(imgs[ch]) * virtual_coil for ch in range(nch)])
+    hanning_1d = np.expand_dims(np.hanning(np.minimum(nx,ny)), 1)
+    hanning_2d = np.fft.fftshift(np.dot(hanning_1d, hanning_1d.T))
+    hanning_Nd = np.tile(np.expand_dims(np.tile(hanning_2d, (nch, 1, 1)), -1), (1, 1, 1, nz))
```

Comment:
Maybe try to generate the hanning filter using a loop here? That way we could be general with dimension.